app/js/components/Workflows.jsx

```diff
@@ -2,24 +2,22 @@ import React from 'react';
 
 import Workflow from './Workflow';
 
-const Workflows = ({ plugin }, { router }) => (
+const Workflows = ({ plugin, openWorkflow }) => (
     <div>
         { plugin.workflows.map(workflow => (
             <Workflow
                 key={workflow.name}
                 flow={workflow}
-                onClick = {() => router.push(`job/${plugin.name}/${workflow.name}`)}
+                onClick = {() => openWorkflow(plugin, workflow)}
             />
         ))}
     </div>
 );
 
 
 Workflows.propTypes = {
-    plugin: React.PropTypes.object
-};
-
-Workflows.contextTypes = {
+    openWorkflow: React.PropTypes.func,
+    plugin: React.PropTypes.object,
     router: React.PropTypes.object
```

Comment:
It is used, but at the container level `mapDispatchToProps = (dispatch, { router }) => {`, so I wasn't sure if it should be left. I left it in `Job.jsx` as well for the same reason. I'll look into if they're needed to be lifted to the container level `ownProps` parameter of the `connect` mapping functions.